official/utils/logs/logger.py

```diff
@@ -246,6 +255,20 @@ def log_run_info(self, model_name, dataset_name, run_params):
          self._bigquery_run_table,
          self._run_id,
          run_info))
+    thread.start_new_thread(
+        self._bigquery_uploader.insert_run_status,
+        (self._bigquery_data_set,
+         self._bigquery_run_status_table,
+         self._run_id,
+         "running"))
+
+  def on_finish(self):
+    thread.start_new_thread(
+        self._bigquery_uploader.update_run_status,
+        (self._bigquery_data_set,
+         self._bigquery_run_status_table,
+         self._run_id,
+         "success"))
```

Comment:
When does this actually get called? Will it have to get called explicitly in each run loop? Any way we can avoid that?

Reply:
Adding the benchmark_context which manage the status update.